content/en/blog/_posts/2023-04-11-kmsv2-beta.md

```diff
@@ -0,0 +1,112 @@
+---
+layout: blog
+title: "Kubernetes 1.27: KMS V2 Moves to Beta"
+date: 2023-04-11
+slug: kms-v2-moves-to-beta
+---
+
+**Authors:** TBD
+
+With Kubernetes 1.27, we are moving Key Management Service (KMS) v2 API to beta.
+
+## What is KMS?
+One of the first things to consider when securing a Kubernetes cluster is encrypting etcd data at rest. KMS provides an interface for a provider to utilize a key stored in an external key service to perform this encryption.
+
+KMS v1 has been a feature of Kubernetes since version 1.10, and is currently in beta as of version v1.12. KMS v2 was introduced as alpha in v1.25.
+
+{{< note >}} The design in [the previous blog post](https://kubernetes.io/blog/2022/09/09/kms-v2-improvements/) is not compatible with the design in this blog post. The KMS v2 API and implementation changed in incompatible ways in-between the alpha release in v1.25 and the beta release in v1.27. Attempting to upgrade from old versions with the alpha feature enabled will result in data loss. {{< /note >}}
```

Comment:
```suggestion
{{< note >}}
The design in [the previous blog post](https://kubernetes.io/blog/2022/09/09/kms-v2-improvements/)
is not compatible with the design in this blog post. The KMS v2 API and implementation changed
in incompatible ways in-between the alpha release in v1.25 and the beta release in v1.27.
Attempting to upgrade from old versions with the alpha feature enabled will result in data loss.
{{< /note >}}
```

Reply:
Ideally, avoid using callouts (note, warning, etc) in blog articles. This is for a couple of reasons, one is maintainability (it's more to redo if we ever change how callouts work) and the other is that blogs aren't docs.

Using plain Markdown, perhaps with an emoji - eg :warning: - is how I'd do it.

---

Site relative links are better still:
```markdown
The design in the [previous blog post](/blog/2022/09/09/kms-v2-improvements/)
is not compatible with the design covered in this blog post.  …
```

Reply:
thanks for the feedback, @sftim! the caution called out here is something we **really** want to ensure is read before using the feature. We also have this in the docs but wanted to echo this in the blog too. Having said that, we're trying to use the emoji option but the current rendered docs with the callouts (https://deploy-preview-40194--kubernetes-io-main-staging.netlify.app/blog/2023/04/11/kms-v2-moves-to-beta/) look much cleaner. Is there any chance we could use it for the blog? The note we might be able to remove the callout and just italicize the text.

Reply:
Try a Docsy [alert](https://www.docsy.dev/docs/adding-content/shortcodes/#alert). There are a few examples of these around the site already.

Reply:
the Docsy alert looks good! thanks for the suggestion